src/datasources.js

```diff
@@ -300,7 +307,14 @@ define([
         }
         function getNodes(source, path, info) {
             var nodes = _.chain(source && source.structure)
-                .map(node(source, path, info))
+                .map(function (item, id) {
+                    if (item.merge) {
```

Comment:
I wasn't expecting it here. Just thinking out loud about how this function is always a little difficult to get spun up on